src/settings/mod.rs

```diff
@@ -1,29 +1,69 @@
 //! Booster NGFW NVM settings
 
-pub mod channel_settings;
+use core::fmt::Write;
+use core::str::FromStr;
+use heapless::String;
+use miniconf::Tree;
+
+pub mod eeprom;
 pub mod flash;
-pub mod global_settings;
 pub mod runtime_settings;
-mod sinara;
-use encdec::{Decode, DecodeOwned, Encode};
-use serde::{Deserialize, Serialize};
 
-use sinara::{BoardId as SinaraBoardId, SinaraConfiguration};
+use eeprom::main_board::IpAddr;
+use runtime_settings::RuntimeSettings;
+use smoltcp_nal::smoltcp;
+
+#[derive(Clone, Debug, Tree)]
+pub struct Settings {
+    #[tree(depth = 4)]
+    pub booster: RuntimeSettings,
+
+    #[tree(skip)]
+    pub mac: smoltcp_nal::smoltcp::wire::EthernetAddress,
+
+    #[tree(validate=Self::validate_ip)]
+    pub ip: String<18>,
```

Comment:
Ideally this would be a `smoltcp::wire::IpCidr`. @jordens can you think of a means that Miniconf could support that?

Reply:
I think get+get_mut+typ might work. Iirc there is a doc test or unit test on that. https://github.com/quartiq/miniconf/blob/5a2e163a9cbc4c8bd396242a146ce43be06f3495/miniconf/tests/validate.rs#L55

Reply:
The problem is `get` and `get_mut` return a `&<typ>`. That doesn't work for i.e. a temporarily-allocated `String<18>` that we print the CIDR into

Reply:
I guess we could store the stringified and unstringified version in the `Settings` struct and skip the string field?

```rs
#[derive(Tree)]
struct Settings {
    #[tree(typ="heapless::String<18>", get=Self::get_ip_str, get_mut=Self::get_ip_str_mut)]
    pub ip: smoltcp::wire::IpCidr,

    #[tree(skip)]
    ip_str: String<18>,
}
```
?

Reply:
Yes. With the caveat that you need to keep them synchronized.
Might be better to just derive Serialize/Deserialize (https://serde.rs/remote-derive.html). Also I'd transition to Ipv4Cidr only here.

Reply:
I must have been tired yesterday. We do the same thing with IP addresses already, so I've reproduced that.